conan/tools/google/toolchain.py

```diff
@@ -1,24 +1,154 @@
+    already available in its bazel_tools repo. For now, it only admits a list of platforms defined
+    by the user.
+    More information related:
+        * Toolchains: https://bazel.build/extending/toolchains (deprecated)
+        * Platforms: https://bazel.build/concepts/platforms (new default since Bazel 7.x)
+        * Migrating to platforms: https://bazel.build/concepts/platforms
+        * Issue related: https://github.com/bazelbuild/bazel/issues/6516
+
+    Others:
+        CROOSTOOL: https://github.com/bazelbuild/bazel/blob/cb0fb033bad2a73e0457f206afb87e195be93df2/tools/cpp/CROSSTOOL
+        Cross-compiling with Bazel: https://ltekieli.com/cross-compiling-with-bazel/
+        bazelrc files: https://bazel.build/run/bazelrc
+        CLI options: https://bazel.build/reference/command-line-reference
+        User manual: https://bazel.build/docs/user-manual
+    """
+
+    bazelrc_name = "conan_bzl.rc"
+    bazelrc_config = "conan-config"
+    bazelrc_template = textwrap.dedent("""
+    # Automatic bazelrc file created by Conan
+    {% if copt %}build:conan-config {{copt}}{% endif %}
+    {% if conlyopt %}build:conan-config {{conlyopt}}{% endif %}
+    {% if cxxopt %}build:conan-config {{cxxopt}}{% endif %}
+    {% if linkopt %}build:conan-config {{linkopt}}{% endif %}
+    {% if force_pic %}build:conan-config --force_pic={{force_pic}}{% endif %}
+    {% if dynamic_mode %}build:conan-config --dynamic_mode={{dynamic_mode}}{% endif %}
+    {% if strip %}build:conan-config --strip={{strip}}{% endif %}
+    {% if compilation_mode %}build:conan-config --compilation_mode={{compilation_mode}}{% endif %}
+    {% if compiler %}build:conan-config --compiler={{compiler}}{% endif %}
+    {% if cpu %}build:conan-config --cpu={{cpu}}{% endif %}
+    {% if crosstool_top %}build:conan-config --crosstool_top={{crosstool_top}}{% endif %}""")
+
+    def __init__(self, conanfile):
         self._conanfile = conanfile
-        self._namespace = namespace
+
+        # Flags
+        # TODO: Should we read the buildenv to get flags?
+        self.extra_cxxflags = []
+        self.extra_cflags = []
+        self.extra_ldflags = []
+        self.extra_defines = []
+
+        # Bazel build parameters
+        shared = self._conanfile.options.get_safe("shared")
+        fpic = self._conanfile.options.get_safe("fPIC")
+        self.force_pic = fpic if (not shared and fpic is not None) else None
+        # FIXME: Keeping this option but it's not working as expected. It's not creating the shared
+        #        libraries at all.
+        self.dynamic_mode = "fully" if shared else "off"
+        self.cppstd = cppstd_flag(self._conanfile.settings)
+        self.copt = []
+        self.conlyopt = []
+        self.cxxopt = []
+        self.linkopt = []
+        self.strip = 'sometimes'  # 'always', 'sometimes', 'never'
+        self.compilation_mode = "fastbuild"  # 'fastbuild', 'dbg', 'opt'
```

Comment:
Checking the docs I think we don't need to define `strip` because it has a default of `sometimes` in Bazel, that will only strip the debug information for `fastbuild` mode and `fastbuild` mode I think that should only be defined by the user because it does not directly correspond to neither Debug or Release of our settings. So I would propose setting something like:

```
self.strip = None
self.compilation_mode = {'Release': 'opt', 
                                            'Debug': 'dbg'}.get(self.settings.build_type)
```

The user will be able to change this in case they have another custom build type defined.